src/Middleware/HttpOverrides/src/CertificateForwarderMiddleware.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpOverrides
+{
+    /// <summary>
+    /// Middleware that converts a forward header into a client certificate if found.
+    /// </summary>
+    public class CertificateForwarderMiddleware
```

Comment:
Why isn;t this part of fowarded headers middleware?

Reply:
We looked at it but the ForwardedHeaders middleware is much more complex. That one handles lists of forwarders and steps through the list. This one only handles a single value.

There's another middleware in this package already that similarly handles a single method override header.

Reply:
Why is this different? I'm not convinced this should be a separate middleware. Don't all of the spame spoofing concerns apply? Why doesn't this have those features? Why are certs different from *all* of the other headers forwarded by a reverse proxy?